server/src/main/java/org/opensearch/index/analysis/NamedAnalyzer.java

```diff
@@ -70,8 +70,8 @@ public NamedAnalyzer(String name, AnalyzerScope scope, Analyzer analyzer, int po
         this.scope = scope;
         this.analyzer = analyzer;
         this.positionIncrementGap = positionIncrementGap;
-        if (analyzer instanceof org.opensearch.index.analysis.AnalyzerComponentsProvider) {
-            this.analysisMode = ((org.opensearch.index.analysis.AnalyzerComponentsProvider) analyzer).getComponents().analysisMode();
+        if (analyzer instanceof org.opensearch.index.analysis.AnalyzerComponentsProvider analyzerComponentsProvider) {
```

Comment:
nit: you don't need to specify the full path name for this class - checked it. It's in the same module, don't even need to import it explicitly also.